src/sentry/api/event_search.py

```diff
@@ -776,7 +776,8 @@ def to_list(value):
                     projects = get_projects(params)
                 condition = ["project_id", "=", projects.get(term.value.value)]
                 kwargs["conditions"].append(condition)
-            elif name == "issue.id":
+            elif name == "issue.id" and term.value.value != "":
```

Comment:
Should you also check the operator? The `!has:issue.id` expression will fall into this case as well.

Reply:
So both `!has:issue.id` and `has.issue.id` have the same name/value, but the operator changes, so both will get excluded by this `elif` here, and caught by the `else` later.

- Added test to show this :+1: